### PR TITLE
docs: edit binding-syntax doc

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

This PR cleans up the `data-binding.md` document.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This PR clarifies content, structure, and headers, moves to ref section. Code in the example is not changed.

**Outstanding question:**

### CODE REVIEW

File: aio/content/guide/binding-syntax.md [hunk 1]
```diff
@@ -1,23 +1,127 @@
...
-# Binding syntax: an overview
+# Binding syntax
 
-Data-binding is a mechanism for coordinating what users see, specifically
-with application data values.
-While you could push values to and pull values from HTML,
-the application is easier to write, read, and maintain if you turn these tasks over to a binding framework.
-You simply declare bindings between binding sources, target HTML elements, and let the framework do the rest.
+Data-binding coordinates the application data values that users see.
+Use data-binding to specify things such as the source of an image, the state of a button, or data for a particular user.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:5]: Marking this here, but it occurs elsewhere (and probably throughout our docs :) ). I think saying "You <verb>" is better than just "<verb>". It reads more casually, and also brings the reader in. So instead of:

Use databinding...

We should say

You use data-binding.

Also, I just realized: do we need to hyphenate the terms data binding and event binding? I don't think it's necessary myself. And editors that I've worked with often say: "When in doubt, remove the hyphen." Thoughts?

File: aio/content/guide/binding-syntax.md [hunk 2]
```diff
@@ -1,23 +1,127 @@
...
 
-Angular provides many kinds of data-binding. Binding types can be grouped into three categories distinguished by the direction of data flow:
 
-* From the _source-to-view_
-* From _view-to-source_
-* Two-way sequence: _view-to-source-to-view_
+## Data-binding and HTML
+
+On its own, HTML modifies elements by setting element attributes with string constants.
+In the following example, `class`, `src`, and `disabled` modify the `<div>`, `<img>`, and `<button>` respectively.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:17]:
```suggestion
In the following example, `class`, `src`, and `disabled` modify the `<div>`, `<img>`, and `<button>` elements respectively.
```

File: aio/content/guide/binding-syntax.md [hunk 3]
```diff
@@ -1,23 +1,127 @@
...
+Use data-binding to control things like the state of a button:
+
+<code-example path="binding-syntax/src/app/app.component.html" region="disabled-button" header="src/app/app.component.html"></code-example>
+
+Notice that the binding is to the `disabled` property of the button's DOM element, not the attribute.
+Data-binding works with properties of DOM elements, components, and directives, not HTML attributes.
+
+{@a html-attribute-vs-dom-property}
+
+### HTML attribute vs. DOM property
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:34]:
```suggestion
### HTML attributes and DOM properties
```
I'm not a fan of `vs.` or `versus`. It's not a competition, after all. :)

File: aio/content/guide/binding-syntax.md [hunk 4]
```diff
@@ -1,23 +1,127 @@
...
+
+The distinction between an HTML attribute and a DOM property is key to understanding
+how Angular binding works.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:37]:
```suggestion
To understand how Angular binding works, you must understand the distinction between an HTML attribute and a DOM property.
```
Rewriting to make the verbs more active and avoid `is`.

Question: Don't we ask our users to have some understanding of HTML and the DOM? 

I bring this up because we don't explain typescript very often, and that' equally critical to using Angular effectively. But we often act as though our users don't understand HTML or the DOM.

File: aio/content/guide/binding-syntax.md [hunk 5]
```diff
@@ -1,23 +1,127 @@
...
+HTML attributes are values that you can configure to modify an element's behavior.
+Browsers create DOM, or Document Object Model, nodes from HTML and these nodes have properties.
+
+* A few HTML attributes have 1:1 mapping to properties; for example, `id`.
+
+* Some HTML attributes don't have corresponding properties; for example, `aria-*`.
+
+* Some DOM properties don't have corresponding attributes; for example, `textContent`.
+
+It is important to remember that HTML attributes and DOM properties are different things, even when they have the same name.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:47]:
```suggestion
<div class="alert is-important">

Remember that HTML attributes and DOM properties are different things, even when they have the same name.

</div>
```

File: aio/content/guide/binding-syntax.md [hunk 6]
```diff
@@ -1,23 +1,127 @@
...
+
+Template binding works with properties and events, not attributes.
+In Angular, the only role of HTML attributes is to initialize element and directive state.
+
+When you write a data-binding, you're dealing exclusively with the DOM properties and events of the target object.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:52]: I'm becoming more and more convinced to remove the hyphen: `data binding`.

File: aio/content/guide/binding-syntax.md [hunk 7]
```diff
@@ -1,23 +1,127 @@
...
+
+<div class="alert is-helpful">
+
+The task of attributes is to initialize DOM properties.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:56]:
```suggestion
Attributes initialize DOM properties.
```

Attempted rewrite for brevity.

File: aio/content/guide/binding-syntax.md [hunk 8]
```diff
@@ -1,23 +1,127 @@
...
+
+Property values can change; attribute values can't, with one exception.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:58]:
```suggestion
Property values can change. Attribute values can't, with one exception:
```

File: aio/content/guide/binding-syntax.md (hunk 7 above)
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:56]: I think this note should be further up (around line 38) where we talk about attributes and properties.

Also, consider removing the note. Users sometimes skip over them. (Even though they shouldn't!)

File: aio/content/guide/binding-syntax.md [hunk 9]
```diff
@@ -1,23 +1,127 @@
...
+You can change attributes with `setAttribute()`, which re-initializes corresponding DOM properties.
+
+</div>
+
+#### Example 1: an `<input>`
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:63]: Four levels of headings is getting awfully deep. Just noting this.

File: aio/content/guide/binding-syntax.md [hunk 10]
```diff
@@ -1,23 +1,127 @@
...
+
+When the browser renders `<input type="text" value="Sarah">`, it creates a
+corresponding DOM node with a `value` property and initializes that `value` to "Sarah".
+
+```html
+<input type="text" value="Sarah">
+```
+
+When the user enters "Sally" into the `<input>`, the DOM element `value` property becomes "Sally".
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:72]: Another note: let's talk about when to use quote marks. I dislike them due to their overuse. (So in this case, "Sally" would become `Sally`).

Not a required change.

File: aio/content/guide/binding-syntax.md [hunk 11]
```diff
@@ -1,23 +1,127 @@
...
+  <input [disabled]="condition ? true : false">
+  <input [attr.disabled]="condition ? 'disabled' : null">
+  ```
+
+  The first line, which uses the `disabled` property, uses a Boolean value.
+  The second line, which uses the disabled attribute checks for `null`.
+
+  Generally, use property binding over attribute binding as a Boolean value is easy to read, the syntax is shorter, and a property is more performant.
+
+</div>
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:111]: I'm wary of large notes like this. Let's consider moving the content into its own section.

File: aio/content/guide/binding-syntax.md [hunk 12]
```diff
@@ -1,23 +1,127 @@
...
+This example shows you how to toggle the disabled property from the component.
+
+
+## Types of data-binding
+
+Angular provides three categories of data-binding according to the direction of data flow:
+
+* From the source-to-view
+* From view-to-source
+* In a two-way sequence of view-to-source-to-view
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:124]:
```suggestion
* From the source to the view
* From the view to the source
* In a two-way sequence of view to source to view
```

File: aio/content/guide/binding-syntax.md [hunk 13]
```diff
@@ -95,125 +199,25 @@ Angular provides many kinds of data-binding. Binding types can be grouped into t
...
-However, if you look at the HTML attribute `value` using `input.getAttribute('value')`, you can see that the *attribute* remains unchanged&mdash;it returns "Sarah".
-
-The HTML attribute `value` specifies the *initial* value; the DOM `value` property is the *current* value.
-
-To see attributes versus DOM properties in a functioning app, see the <live-example name="binding-syntax"></live-example> especially for binding syntax.
 
-#### Example 2: a disabled button
+Binding types other than interpolation have a target name to the left of the equal sign.
+The target of a binding is a property or event, which you surround with square brackets, `[]`, parentheses, `()`, or both, `[()]`.
+Alternatively, you can specify a target with the prefix `bind-`, `on-`, or `bindon-`.
```
Comment (aikidave) [inline aio/content/guide/binding-syntax.md:205]: Maybe delete this line. Let's highlight the preferred way of doing this?

**Review by aikithoughts — COMMENTED** [review]
A few thoughts! Thanks for doing this.

Also, can you update the navigation.json file for this PR so this topic appears in the Conceptual Reference section?

File: aio/content/guide/binding-syntax.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:5]: Ok, switching from imperative to indicative.

Regarding hyphens, I'm all for leaving them out when possible, especially when they don't clarify meaning.

File: aio/content/guide/binding-syntax.md (hunk 4 above)
Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:37]: Agreed. DOM, properties, and attributes seem foundational concepts that already have excellent free resources. There was some concern at a point a couple years ago about readers not understanding these concepts and we started explaining them, which may occasionally be appropriate if it's something obscure, but I agree that we don't have to do things like explain what DOM means each time we use it.

Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:37]: I changed this section to read:

```
Angular binding distinguishes between HTML attributes and DOM properties.

HTML attributes are values that you can configure to modify an element's behavior.
Properties are features of DOM nodes.
```

What do you think? Still too granular?

File: aio/content/guide/binding-syntax.md (hunk 7 above)
Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:56]: Moved up and incorporated into the copy about attributes and properties. Removed alert styling.

File: aio/content/guide/binding-syntax.md (hunk 9 above)
Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:63]: `<h4>`s also don't show up in the TOC.

Comment (aikidave) [reply aio/content/guide/binding-syntax.md:63]: Let's consider using level 4 headings with caution. But I don't think this needs to hold up this PR.

**Review by aikithoughts — APPROVED** [review]

File: aio/content/guide/binding-syntax.md [hunk 14]
```diff
@@ -1,23 +1,122 @@
 
-# Binding syntax: an overview
+# Binding syntax
 
-Data-binding is a mechanism for coordinating what users see, specifically
-with application data values.
-While you could push values to and pull values from HTML,
-the application is easier to write, read, and maintain if you turn these tasks over to a binding framework.
-You simply declare bindings between binding sources, target HTML elements, and let the framework do the rest.
+Data binding coordinates the application data values that users see.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:4]: How about
```suggestion
Data binding automatically keeps your page up-to-date based on your application's state.
```

File: aio/content/guide/binding-syntax.md [hunk 15]
```diff
@@ -1,23 +1,122 @@
...
 </div>
 
-Angular provides many kinds of data-binding. Binding types can be grouped into three categories distinguished by the direction of data flow:
 
-* From the _source-to-view_
-* From _view-to-source_
-* Two-way sequence: _view-to-source-to-view_
+## Data binding and HTML
+
+On its own, HTML modifies elements by setting element attributes with string constants.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:16]: The phrase "HTML modifies elements" is a bit off since "HTML" isn't the party performing the action. Instead it would be something along the lines of "Developers can customize HTML by specifying attributes with string values."

File: aio/content/guide/binding-syntax.md [hunk 16]
```diff
@@ -1,23 +1,122 @@
...
+
+* A few HTML attributes have 1:1 mapping to properties; for example, `id`.
+
+* Some HTML attributes don't have corresponding properties; for example, `aria-*`.
+
+* Some DOM properties don't have corresponding attributes; for example, `textContent`.
+
+Property values can change.
+Attribute values can't, with one exception:
+You can change attributes with `setAttribute()`, which re-initializes corresponding DOM properties.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:49]: I might omit this bit. In reality, both properties and attributes can be changed at any time, it's just that the way you change them is different (directly setting properties vs. using `setAttribute` or `removeAttribute`). In many cases, setting _either_ the attribute or the property will update the other (for tabindex, id, name, disabled, checked, readonly, and probably more)

File: aio/content/guide/binding-syntax.md [hunk 17]
```diff
@@ -1,23 +1,122 @@
...
+
+
+<div class="alert is-important">
+
+Remember that HTML attributes and DOM properties are different things, even when they have the same name.
+
+</div>
+
+Template binding works with properties and events, not attributes.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:58]: This statement isn't 100% accurate; you can set up a binding on an attribute, it just uses a slightly different syntax (`[attr.whatever]` instead of `[whatever]`)

File: aio/content/guide/binding-syntax.md [hunk 18]
```diff
@@ -95,125 +194,24 @@ Angular provides many kinds of data-binding. Binding types can be grouped into t
...
-
-
-To see the `disabled` button example in a functioning app, see the <live-example name="binding-syntax"></live-example> especially for binding syntax. This example shows you how to toggle the disabled property from the component.
 
 ## Binding types and targets
 
-The **target of a data-binding** is something in the DOM.
-Depending on the binding type, the target can be a property (element, component, or directive),
-an event (element, component, or directive), or sometimes an attribute name.
+The target of a data binding can be a property, an event, or sometimes an attribute name.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:214]: I would omit the "sometimes" here.

File: aio/content/guide/binding-syntax.md [hunk 19]
```diff
@@ -316,3 +314,6 @@ The following table summarizes the targets for the different binding types.
   </tr>
 </table>
 
+<!-- What is a "public member of a source directive" in plain English? And where does it make sense to put this?  -->
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:317]: "Public member" is a pretty standard term. A member is a property or method on the class, and public means that it's part of the public API of that class (not `private` or `protected`).

File: aio/content/guide/binding-syntax.md (hunk 17 above)
Comment (kapunahelewong) [reply aio/content/guide/binding-syntax.md:58]: Thanks. I've removed this sentence.

File: aio/content/guide/binding-syntax.md [hunk 20]
```diff
@@ -1,23 +1,116 @@
...
+
+Though you could technically set the `[attr.disabled]` attribute binding, the values are different in that the property binding must be a boolean value, while its corresponding attribute binding relies on whether the value is `null` or not.
+Consider the following:
+
+```html
+<input [disabled]="condition ? true : false">
+<input [attr.disabled]="condition ? 'disabled' : null">
+```
+
+The first line, which uses the `disabled` property, uses a Boolean value.
```
Comment (jelbourn) [inline aio/content/guide/binding-syntax.md:98]:
```suggestion
The first line, which uses the `disabled` property, uses a boolean value.
```
boolean typically wouldn't be capitalized when talking about javascript/typescript (here and elsewhere)

**Review by jelbourn — APPROVED** [review]
LGTM, one last tiny comment

**Review by jessicajaniuk — APPROVED** [review]
